swagger.yaml

```diff
@@ -5190,6 +5203,22 @@ definitions:
       finalizing_stage_percentage:
         type: integer
 
+  disk-encryption:
+    type: object
+    properties:
+      enable_on:
+        type: string
+        description: Enable/disable disk encryption on master nodes, worker nodes, or all nodes.
+        enum: ['none', 'all', 'masters', 'workers']
+      mode:
+        type: string
+        description: The disk encryption mode to use.
+        enum: ['tpmv2', 'tang']
+      tang_servers:
+        type: string
+        description: JSON-formatted string containing additional information regarding tang's configuration
```

Comment:
Does this follow the ignition {url, thumbprint} convention, or the clevis {url, thp} convention?

Would it be better to just have the spec ask for an array of {url, thumbprint} pairs to make it unambiguous and less prone to typos or json-in-a-string-in-yaml escaping errors?

Reply:
> Does this follow the ignition {url, thumbprint} convention, or the clevis {url, thp} convention

It follows the ignition, you can see it here https://coreos.github.io/ignition/configuration-v3_2/

> Would it be better to just have the spec ask for an array of {url, thumbprint} pairs

This will require maintaining 2 different objects in swagger - 1 for the cluster object and one for create/update cluster params.

The reason for that is that we don't know the length of this list, therefore, the ways to store it in DB are by Marshaling it to a json-string or to have a separate table in DB for them.

The second option seems a bit of overkill to me, so I decided to go with the first option as we do for `validations_info`.

If you have a better option I am open for suggestions.
@lack

Reply:
It definitely meets the requirements for what NBDE needs, and given the marshalling requirement makes sense?  Unless something can let you de-couple the data format here from that implementation detail?  I don't know enough about your architecture to say!  I wasn't sure if the consumers of your API are administrators creating CRs or other non-human actors, so I worry about the potential errors of having to pass structured data via an unstructured string.

Maybe we should just make sure the description is clear about what the format is, since there's are a couple different possibilities of how this data may legitimately be structured?

Clevis:
```
[{"url": "http://host.example.com:7500”, "thp": "xyz"},...]
```

Ignition:
```
[{"url": "http://host.example.com:7500”, "thunbprint": "xyz"},...]
```

Your example of capitalized ignition:
```
[{"URL": "http://host.example.com:7500”, "Thumbprint": "xyz"},...]
```

Reply:
This was just an example a got when playing with the template, in the code they are called `url` and `thumbprint`

Reply:
Fixed.